src/interceptors/XMLHttpRequest/XMLHttpRequestController.ts

```diff
@@ -769,3 +767,8 @@ async function getXMLHttpRequestBodyInitLength(
 
   return body.toString().length
 }
+
+async function inferResponseBodyLength(response: Response): Promise<number> {
+  const buffer = await response.clone().arrayBuffer();
```

Comment:
This is so nice I'm considering using something like this for calculating request body length as well. As long as the environment will construct the correct `Request`, it will take account things like FormData boundaries, which we are calculating manually right now.

Reply:
Thanks! I'm really glad you liked the approach. I'd be happy to help if you want to apply this method elsewhere. Let me know!